pages/organizations/[organizationSlug]/activity.tsx

```diff
@@ -0,0 +1,328 @@
+import {
+  Activity as ActivityTypePrisma,
+  DataSource,
+  User,
+  View,
+} from "@prisma/client";
+import {
+  ArrowRightIcon,
+  PlusIcon,
+  RefreshIcon,
+  TrashIcon,
+} from "@heroicons/react/outline";
+import { Avatar, AvatarBadge, Code, Tooltip } from "@chakra-ui/react";
+import { DateTime } from "luxon";
+import { isEmpty, isNull, isUndefined } from "lodash";
+import { useGetActivitiesQuery } from "@/features/organizations/api-slice";
+import { useOrganizationFromProfile } from "@/hooks";
+import { useRouter } from "next/router";
+import Layout from "@/components/Layout";
+import LoadingOverlay from "@/components/LoadingOverlay";
+import OffsetPaginationComponent from "@/features/tables/components/OffsetPaginationComponent";
+import OrganizationSidebar from "@/components/OrganizationSidebar";
+import PageWrapper from "@/components/PageWrapper";
+import React, { useMemo, useState } from "react";
+import md5 from "md5";
+
+export type ActivityType = ActivityTypePrisma & {
+  dataSource?: DataSource;
+  view?: View;
+  user: User;
+};
+
+const ActivityItem = ({
+  activity,
+  lastItem = false,
+}: {
+  activity: ActivityType;
+  lastItem?: boolean;
+}) => {
+  const avatarIcon = useMemo(() => {
+    switch (activity.action) {
+      case "create":
+        return <PlusIcon className="w-3 h-3 text-white" />;
+      case "bulkDelete":
+      case "delete":
+        return <TrashIcon className="w-3 h-3 text-white" />;
+      case "update":
+        return <RefreshIcon className="w-3 h-3 text-white" />;
+    }
+  }, [activity.action]);
+
+  const avatarBadgeColor = useMemo(() => {
+    switch (activity.action) {
+      case "create":
+        return "green";
+      case "bulkDelete":
+      case "delete":
+        return "red";
+      case "update":
+        return "blue";
+    }
+  }, [activity.action]);
+
+  const urlToSource = useMemo(
+    () =>
+      activity?.view
+        ? `/views/${activity.view.id}`
+        : `/data-sources/${activity?.dataSource?.id}/tables/${activity.tableName}`,
+    [activity?.view, activity?.dataSource, activity.tableName]
+  );
+
+  const urlToRecord = useMemo(
+    () =>
+      activity?.view
+        ? `${urlToSource}/records/${activity.recordId}`
+        : `${urlToSource}/${activity.recordId}`,
+    [urlToSource, activity.recordId]
+  );
+
+  const sourceName = useMemo(
+    () =>
+      activity?.view
+        ? activity?.view.name
+        : activity.dataSource?.name + "/" + activity.tableName,
+    [activity?.view, activity?.dataSource, activity.tableName]
+  );
+
+  const message = useMemo(() => {
+    switch (activity.action) {
+      case "create":
+        return (
+          <p>
+            Created record{" "}
+            <Code>
+              <a href={urlToRecord}>{activity.recordId}</a>
+            </Code>{" "}
+            in{" "}
+            <Code>
+              <a href={urlToSource}>{sourceName}</a>
+            </Code>
+          </p>
+        );
```

Comment:
Can we make all of these components? 
Like `<CreateMessage />`, `<BulkDeleteMessage />` etc? And use this switch to choose from those components?

It's a bit weird mixing this much "HTML" in "code".